adapter: add mz_subscribes system table (#17906)
MaterializeInc/materialize: doc/user/content/sql/system-catalog/mz_internal.md

`index_id` | [`text`]   | The ID of the index the peek is targeting.
`time`     | [`mz_timestamp`] | The timestamp the peek has requested.

### `mz_subscribes`

The `mz_subscribes` table describes all currently in-progress
[`SUBSCRIBE`](/sql/subscribe) operations.

Field        | Type                         | Meaning
-------------|------------------------------|--------

> `id`         | [`text`]                     | The ID of the subscribe.
> `user`       | [`text`]                     | The user who started the subscribe.
> `cluster_id` | [`text`]                     | The ID of the cluster on which the subscribe is running. Corresponds to [`mz_clusters.id`](../mz_catalog/#mz_clusters).
> `started_at` | [`timestamp with time zone`] | The time at which the subscribe was started.

Reviewer: ```suggestion
`id`         | [`text`]                     | The ID of the subscription.
`user`       | [`text`]                     | The user who started the subscription.
`cluster_id` | [`text`]                     | The ID of the cluster maintaining the subscription. Corresponds to [`mz_clusters.id`](../mz_catalog/#mz_clusters).
`started_at` | [`timestamp with time zone`] | The time at which the subscription was started.
```

Reviewer: We should either use "`SUBSCRIBE` operation" or "subscription".

Reviewer: Oh, this makes me wonder if the table should actually be called `mz_subscriptions`?

Author: I think Nikhil at some point raised the idea that "subscriptions" should be permanent subscribe. So you'd have:
* `SUBSCRIBE t` which aborts when you end the session
* `CREATE SUBSCRIPTION ON t` which runs until it is `DROP`ed

So there is a possible future in which we'd have both of these things. That said, this is just a musing for now and the table lives in `mz_internal` anyway, so we can change the name later. I'll rename it to `mz_subscriptions`.

Reviewer: Thanks for digging that up, I'd forgotten about the discussion! ISTM like this could then be reflected as a flag that indicates whether the subscription is temporary or not. If we ever make that distinction, it's a bit awkward to not use `CREATE TEMP(ORARY) SUBSCRIPTION` (since we use that syntax for _e.g._ temporary table creation), though I get it'd introduce more overhead than just typing `SUBSCRIBE`.
